nvme.c

```diff
@@ -2400,8 +2404,16 @@ static int create_ns(int argc, char **argv, struct command *cmd, struct plugin *
 		goto close_fd;
 	}
 
-	nvme_init_id_ns(&ns, cfg.nsze, cfg.ncap, cfg.flbas, cfg.dps, cfg.nmic,
-			 cfg.anagrpid, cfg.nvmsetid);
+	memset(&ns, 0, sizeof(ns));
```

Comment:
Let's keep it consistent with the rest. For this kind of initialization we use following pattern in the rest of the file:

```
	struct nvme_io_args args = {
		.args_size	= sizeof(args),
		.fd		= fd,
		.nsid		= cfg.namespace_id,
		.slba		= cfg.start_block,
		.nlb		= cfg.block_count,
		.control	= control,
		.dsm		= cfg.dsmgmt,
		.dspec		= cfg.dspec,
		.reftag		= cfg.ref_tag,
		.apptag		= cfg.app_tag,
		.appmask	= cfg.app_tag_mask,
		.storage_tag	= cfg.storage_tag,
		.data_len	= buffer_size,
		.data		= buffer,
		.metadata_len	= cfg.metadata_size,
		.metadata	= mbuffer,
		.timeout	= NVME_DEFAULT_IOCTL_TIMEOUT,
		.result		= NULL,
	};
```

Reply:
I thought that pattern only works for initialization, but the memset to zero all other fields would need to be first here? Though the memset might not be necessary, it seems that structs initialized that way have all non-specified fields zeroed out anyways by the standard.

Reply:
Hmm, the ns struct is used earlier in the function for the identify command so initializing it here would require having a separate nvme_id_ns. Seems better to just leave it as-is to me, thoughts?

Reply:
We are already exploiting the fact that we switched to C99 and hence we can declare new variable mixed in the code, although only in few places. The general rule is: define all variables at the beginning of the block. But for things like this it's just fine and makes the code more readable.

So just introduce a new `nvme_id_ns` variable right before the `nvme_ns_mgmt_create` call. 

e.g.
```
	struct nvme_io_args args = {
		.args_size	= sizeof(args),
		.fd		= fd,
		.nsid		= cfg.namespace_id,
		.slba		= cfg.start_block,
		.nlb		= cfg.block_count,
		.control	= control,
		.reftag		= cfg.ref_tag,
		.apptag		= cfg.app_tag,
		.appmask	= cfg.app_tag_mask,
		.storage_tag	= cfg.storage_tag,
		.timeout	= NVME_DEFAULT_IOCTL_TIMEOUT,
		.result		= NULL,
	};
	err = nvme_verify(&args);
```

Reply:
Sure, change has been implemented.